docs/Authenticating-and-Authorizing-Users-via-keycloak.md

```diff
@@ -252,6 +253,85 @@ If all goes well, the following should happen:
+
+![](images/previews/oauth2_client_1.png)
+
+2. On the configuration page of `cbioportal_api` client apply the following settings:
+
+#### Settings tab
+
+| parameter        | value  | comment  |
+| ------------- |:-------------:| -----:|
+| Access Type      | confidential |  |
+| Standard Flow Enabled      | ON      |   (default value) |
+| Enabled      | ON      |   (default value) |
+| Consent Required      | OFF        |   (default value) |
+| Client Protocol      | openid-connect   |   (default value) |
+| Access Type | confidential      |     (default value) |
+| Standard Flow Enabled | ON      |     (default value) |
+| Implicit Flow Enabled | OFF      |     (default value) |
+| Direct Access Grants Enabled | OFF      |    |
+| Service Accounts Enabled | OFF      |  (default value)   |
+| Authorization Enabled | OFF      |  (default value)  |
+| Valid Redirect URIs | _url_/api/data-access-token/oauth2  |  _url_ refers to base url of cBioPortal instance |
+
+![](images/previews/oauth2_client_3.png)
+
+#### Credentials tab
+Select `Client Id and Secret`. Take notice of the value of _Secret_ the secret field. This secret should be added to `portal.properties` file of the cBioPortal backend.
+
+| parameter        | value  | comment  |
+| ------------- |:-------------:| -----:|
+| Client Authenticator     | Client Id and Secret |   (default value) |
+
+![](images/previews/oauth2_client_6.png)
+
+#### Client Scopes tab
+Keep only scopes `roles` and `offline_access` (remove all others).
+
+![](images/previews/oauth2_client_2.png)
+
+#### Mapper tab
+Create a new _Audience_ mapper with name `cbioportal_api_audience`. This value will be used by the cBioPortal backend during validation of access tokens.
+
+| parameter        | value  | comment  |
+| ------------- |:-------------:| -----:|
+| Name       | cbioportal_api_audience |  |
+| Mapper Type     | Audience      |   |
+| Included Client Audience      | cbioportal_api      |    |
+| Add to ID token      | OFF        |  (default value)  |
+| Add to access token  | ON      |   (default value)  |
+
+![](images/previews/oauth2_client_4.png)
+
+#### Scope tab
+Enable _Full Scope_. This setting will include the user roles defined in the `cbioportal` SAML client in access tokens distributed by the `cbioportal_api` client.
+
+| parameter        | value  | comment  |
+| ------------- |:-------------:| -----:|
+| Full Scope Allowed       | ON | (default value) |
+
+![](images/previews/oauth2_client_5.png)
+
+3. Add these parameters to `portal.properties` of the cBioPortal backend .
+
+| parameter        | value  | comment  |
+| ------------- |:-------------:| -----:|
+| dat.oauth2.clientId       | cbioportal_api |  |
+| dat.oauth2.clientSecret    | ?      | see _Secret_ field in the _Credentials_ tab  |
+| dat.oauth2.accessTokenUri     | _url_/auth/realms/cbio/protocol/openid-connect/token      |   _url_ refers to base url of the KeyCloak server |
```

Comment:
In this case isn't cbio is the name of the realm you created? I think the docs use cbioportal, but maybe you should use _realm-name_ or something along those lines.  Relevant for some values below...

Reply:
Well spotted. I have updated `/cbio/` to `/cbioportal/`.